backend/src/db/pagination.rs

```diff
@@ -0,0 +1,115 @@
+//! Paginate queries.
+
+use crate::model::dto::Page;
+use diesel::pg::Pg;
+use diesel::query_builder::{AstPass, Query, QueryFragment};
+use diesel::sql_types::{BigInt, Integer};
+use diesel::{QueryId, QueryResult};
+use diesel_async::methods::LoadQuery;
+use diesel_async::{AsyncPgConnection, RunQueryDsl};
+use std::cmp::max;
+
+/// The default number of results for a paginated query.
+const DEFAULT_PER_PAGE: i32 = 10;
+/// The minimum value for page number in a paginated query.
+const MIN_PAGE: i32 = 1;
+/// The minimum value for `per_page` in a paginated query.
+const MIN_PER_PAGE: i32 = 1;
```

Comment:
As the author of this, I'm blind to what needs clarification here.
Can you tell me what was missing when you first read it?

Reply:
Maybe add a short example. 

So as far as I understood `DEFAULT_PER_PAGE` means that you return 10 'rows' when requesting by default.
This makes sense as a variable as you might want to change it later on.

`MIN_PAGE` means that you cannot fetch a page smaller than 1.
`MIN_PER_PAGE` means that you cannot fetch less than 1 rows per page.
For those two (if I understood their usage correctly) I don't think you should use variables as variables signal that you might want to change those values later on. However as far as I understood those values should always be 1 as other values don't really make much sense.

Reply:
I updated it a little.

> I don't think you should use variables as variables signal that you might want to change those values later on.

Shouldn't it be clear that they won't change since they are `const`?
Of course, I could just use a literal `1` in the `per_page` and `paginate` functions but wouldn't that lose more meaning?

Reply:
I like the style of using const variables to avoid "magic constants".